indico/web/client/js/utils/positioning.js

```diff
@@ -125,53 +205,84 @@ const horizontalArrowPosition = {
 };
 
 export const verticalTooltipPositionStrategy = {
+  ...geometry,
   ...verticalPreferAbovePosition,
-  ...unaligned,
+  ...horizontalCenter,
   ...withArrow,
   ...verticalArrowPosition,
 };
 
 export const horizontalTooltipPositionStrategy = {
+  ...geometry,
   ...horizontalTargetPosition,
-  ...unaligned,
+  ...verticalCenter,
   ...withArrow,
   ...horizontalArrowPosition,
 };
 
 export const dropdownPositionStrategy = {
+  ...geometry,
   ...verticalPreferBelowPosition,
   ...horizontalFlush,
   ...withoutArrow,
 };
 
 export const popupPositionStrategy = {
+  ...geometry,
   ...verticalPreferAbovePosition,
   ...unaligned,
   ...withoutArrow,
 };
 
 /**
- * Calculate whether the target element will fit the space below the anchor and apply position styling
+ * Calculate whether the target element will fit the space around the
+ * anchor and apply position styling to clear the anchor.
+ *
+ * The exact strategy for calculating these values is defined by the
+ * `*Strategy` objects defined in this module.
  */
 export function position(target, anchor, strategy, callback) {
   strategy = Object.create(strategy);
   strategy.target = target;
   strategy.anchor = anchor;
 
+  const positioningAbortController = new AbortController();
+  const adjustPosition = () => {
+    strategy.setPosition();
+    strategy.setAlignment();
+    strategy.setArrowDirection();
+  };
+
+  window.addEventListener(
+    'resize',
+    () => {
+      strategy.resetGeometry(adjustPosition);
```

Comment:
Shouldn't this be debounced? Running an event handler on resize cause a pretty high amount of events if you actually resize the window (and not just move e.g. between normal and maximized).

Reply:
Technically it would need to be throttled, not debounced. But it's probably optimizing for an edge case to degrade the experience for the primary case where you are dealing with one-time screen rotate, maximize-minimize type of events.